flow_apps/Flow-Based Knowledge/force-app/main/default/classes/KnowledgeRecommendationsGenerator.cls

```diff
@@ -0,0 +1,91 @@
+public with sharing class KnowledgeRecommendationsGenerator 
+{
+    @InvocableMethod
+    public static List<List<Recommendation>> generateKnowledgeRecommendations(List<Requests> requests)
+    {
+        List<List<Recommendation>> recommendationsResult = new List<List<Recommendation>>();
+        List<Recommendation> recommendations = new List<Recommendation>();
+
+        String searchString = requests[0].searchString;
```

Comment:
the best practice for creating invocable actions is to bulkify them. that means that instead of just taking the first member, with [0], you iterate through the list. Most of the actions in the flowcomponents repo are properly bulkified like this. Here's one example: https://github.com/alexed1/LightningFlowComponents/blob/master/flow_action_components/AddQuotesToFields/force-app/main/default/classes/AddQuotesToFields.cls

When an action is bulkified, it can be used in flows that trigger on batches of records, like a bulk upload of Contacts, and the action will properly handle all of the records at once.

Reply:
resolved it